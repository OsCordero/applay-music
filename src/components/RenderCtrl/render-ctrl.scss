.spinner-container {
  position: relative;
  height: 30vh;
  .spinner {
    position: absolute;
    top: 50%;
    left: 50%;
  }
}
