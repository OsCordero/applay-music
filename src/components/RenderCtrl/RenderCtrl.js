import React from 'react';
import { Alert, Spin } from 'antd';

import './render-ctrl.scss';
export const RenderCtrl = props => {
  return props.error ? (
    <Alert
      message='Something went wrong'
      closable
      description={"We're sorry, please try refresh the page"}
      type='error'
      showIcon
    />
  ) : props.loading ? (
    <div className='spinner-container'>
      <Spin className='spinner' size='large' />
    </div>
  ) : (
    <>{props.children}</>
  );
};

export default RenderCtrl;
