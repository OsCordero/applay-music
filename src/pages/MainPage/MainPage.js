import React, { useEffect, useState } from 'react';
import { Typography, Card, Icon, Pagination, Row, Col } from 'antd';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import { fetchUserAlbums } from 'actions/albumActions';
import RenderCtrl from 'components/RenderCtrl/RenderCtrl';

import './main.scss';
const { Meta } = Card;
const { Title } = Typography;

export const MainPage = props => {
  const { fetchUserAlbums } = props;

  useEffect(() => {
    fetchUserAlbums(0);
  }, [fetchUserAlbums]);

  const [currentPage, setCurrentPage] = useState(1);
  const onPageChange = page => {
    setCurrentPage(page);
    fetchUserAlbums((page - 1) * 4);
  };
  return (
    <div>
      <Title level={2}>Your saved Albums:</Title>
      <RenderCtrl error={props.error} loading={props.loading}>
        <Row type='flex' gutter={[30, 16]}>
          {props.albums.map(albumsItem => {
            const { album } = albumsItem;
            return (
              <Col xs={24} md={12} lg={8} xl={6} key={album.id}>
                <Card
                  hoverable={true}
                  cover={<img alt='album-cover' src={album.images[1].url} />}
                  actions={[
                    <Link to={`/album-detail/${album.id}`} key='edit'>
                      <div>
                        <Icon type='file-search' />
                        See Details
                      </div>
                    </Link>,
                    <a
                      key='ellipsis'
                      target='_blank'
                      rel='noopener noreferrer'
                      href={album.external_urls.spotify}
                    >
                      <Icon type='ellipsis' />
                      Go to spotify
                    </a>,
                  ]}
                >
                  <Link to={`/album-detail/${album.id}`} key='edit'>
                    <Meta title={album.name} description={album.artists[0].name} />
                  </Link>
                </Card>
              </Col>
            );
          })}
        </Row>

        <div className='pagination-container'>
          <Pagination
            pageSize={4}
            style={{ margin: '0 auto' }}
            total={props.total}
            onChange={onPageChange}
            current={currentPage}
          />
        </div>
      </RenderCtrl>
    </div>
  );
};

const mapStateToProps = state => {
  return {
    albums: state.album.albumList,
    total: state.album.totalAlbums,
    error: state.album.albumError,
    loading: state.album.isLoading,
  };
};
export default connect(mapStateToProps, { fetchUserAlbums })(MainPage);
