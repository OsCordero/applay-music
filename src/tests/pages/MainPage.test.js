import React from 'react';
import { shallow } from 'enzyme';
import { MainPage } from 'pages/MainPage/MainPage';
import { RenderCtrl } from 'components/RenderCtrl/RenderCtrl';
import albums from 'tests/fixtures/albums';

test('should render main Page with albums list', () => {
  const wrapper = shallow(<MainPage albums={albums} total={7} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render main Page with error alert on error', () => {
  const wrapper = shallow(
    <RenderCtrl error={true}>
      <MainPage albums={[]} />
    </RenderCtrl>
  );
  expect(wrapper).toMatchSnapshot();
});

test('should render main Page with spinner on loading', () => {
  const wrapper = shallow(
    <RenderCtrl loading={true}>
      <MainPage albums={[]} />
    </RenderCtrl>
  );
  expect(wrapper).toMatchSnapshot();
});
