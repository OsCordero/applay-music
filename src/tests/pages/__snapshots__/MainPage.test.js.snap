// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render main Page with albums list 1`] = `
<div>
  <Title>
    Your saved Albums:
  </Title>
  <RenderCtrl>
    <Row
      gutter={
        Array [
          30,
          16,
        ]
      }
      type="flex"
    >
      <Col
        key="3rxF05Aux0QTrN533Kjc91"
        lg={8}
        md={12}
        xl={6}
        xs={24}
      >
        <Card
          actions={
            Array [
              <ForwardRef
                to="/album-detail/3rxF05Aux0QTrN533Kjc91"
              >
                <div>
                  <Icon
                    type="file-search"
                  />
                  See Details
                </div>
              </ForwardRef>,
              <a
                href="https://open.spotify.com/album/3rxF05Aux0QTrN533Kjc91"
                rel="noopener noreferrer"
                target="_blank"
              >
                <Icon
                  type="ellipsis"
                />
                Go to spotify
              </a>,
            ]
          }
          cover={
            <img
              alt="album-cover"
              src="https://i.scdn.co/image/ab67616d00001e022905ed19f009bd5661019c5e"
            />
          }
          hoverable={true}
        >
          <Link
            key="edit"
            to="/album-detail/3rxF05Aux0QTrN533Kjc91"
          >
            <Meta
              description="MotÃ¶rhead"
              title="Ace of Spades (Expanded Edition)"
            />
          </Link>
        </Card>
      </Col>
      <Col
        key="4otkd9As6YaxxEkIjXPiZ6"
        lg={8}
        md={12}
        xl={6}
        xs={24}
      >
        <Card
          actions={
            Array [
              <ForwardRef
                to="/album-detail/4otkd9As6YaxxEkIjXPiZ6"
              >
                <div>
                  <Icon
                    type="file-search"
                  />
                  See Details
                </div>
              </ForwardRef>,
              <a
                href="https://open.spotify.com/album/4otkd9As6YaxxEkIjXPiZ6"
                rel="noopener noreferrer"
                target="_blank"
              >
                <Icon
                  type="ellipsis"
                />
                Go to spotify
              </a>,
            ]
          }
          cover={
            <img
              alt="album-cover"
              src="https://i.scdn.co/image/3f505feb301e37af5d6e23bc6782bbf6cb575183"
            />
          }
          hoverable={true}
        >
          <Link
            key="edit"
            to="/album-detail/4otkd9As6YaxxEkIjXPiZ6"
          >
            <Meta
              description="Eminem"
              title="Music To Be Murdered By"
            />
          </Link>
        </Card>
      </Col>
    </Row>
    <div
      className="pagination-container"
    >
      <Pagination
        current={1}
        onChange={[Function]}
        pageSize={4}
        style={
          Object {
            "margin": "0 auto",
          }
        }
        total={7}
      />
    </div>
  </RenderCtrl>
</div>
`;

exports[`should render main Page with error alert on error 1`] = `
<Alert
  closable={true}
  description="We're sorry, please try refresh the page"
  message="Something went wrong"
  showIcon={true}
  type="error"
/>
`;

exports[`should render main Page with spinner on loading 1`] = `
<div
  className="spinner-container"
>
  <Spin
    className="spinner"
    size="large"
    spinning={true}
    wrapperClassName=""
  />
</div>
`;
